### PR TITLE
contracts-bedrock: erc721 bridge single impl

### PR DESCRIPTION
**Pull request by tynes**
**Description**

See this [notion doc](https://oplabs.notion.site/External-Reviewing-the-Single-Implementation-Smart-Contract-Changes-a79827afc91c40f7ae13b6340ab84622) on how to review

Closes CLI-4312

### CODE REVIEW

**Comment by tynes** [comment]
This still needs some fixing up

File: packages/contracts-bedrock/test/L2ERC721Bridge.t.sol [hunk 1]
```diff
@@ -2,12 +2,13 @@
...
 
 // Testing utilities
-import { Messenger_Initializer } from "./CommonTest.t.sol";
+import { ERC721Bridge_Initializer } from "./CommonTest.t.sol";
 
 // Target contract dependencies
 import { ERC721 } from "@openzeppelin/contracts/token/ERC721/ERC721.sol";
 import { L1ERC721Bridge } from "../src/L1/L1ERC721Bridge.sol";
 import { OptimismMintableERC721 } from "../src/universal/OptimismMintableERC721.sol";
+import { Proxy } from "../src/universal/Proxy.sol";
```
Comment (refcell) [inline packages/contracts-bedrock/test/L2ERC721Bridge.t.sol]: This doesn't seem used.
```suggestion
```

**Review by refcell — CHANGES_REQUESTED** [review]
Looks good, just one small nit.

**Checks**

Checked L2 Output
```
INFO [08-08|10:47:27.930] L2ERC721Bridge                           implementation=0xC0D3c0d3c0d3c0d3c0D3C0d3C0D3C0D3c0d30014
INFO [08-08|10:47:27.932] L2ERC721Bridge                           MESSENGER=0x4200000000000000000000000000000000000007
INFO [08-08|10:47:27.932] L2ERC721Bridge                           OTHERBRIDGE=0xb2Db8a261CD2c462E3b9548324D5dACf2B7a5687
INFO [08-08|10:47:27.933] L2ERC721Bridge version                   version=1.3.0
```

Implementation messenger correctly zeroed in the devnet
```sh
$ cast call 0xC0D3c0d3c0d3c0d3c0D3C0d3C0D3C0D3c0d30014 \
    "messenger()" \
    --rpc-url http://127.0.0.1:9545
0x0000000000000000000000000000000000000000000000000000000000000000
```

Proxy messenger set correctly
```sh
$ cast call 0x4200000000000000000000000000000000000014 \
    "messenger()" \
    --rpc-url http://127.0.0.1:9545
0x0000000000000000000000004200000000000000000000000000000000000007
```

File: packages/contracts-bedrock/test/L2ERC721Bridge.t.sol (hunk 1 above)
Comment (tynes) [reply packages/contracts-bedrock/test/L2ERC721Bridge.t.sol]: Good catch, fixed, thank you

File: packages/contracts-bedrock/src/L1/L1ERC721Bridge.sol [hunk 2]
```diff
@@ -15,14 +17,17 @@ contract L1ERC721Bridge is ERC721Bridge, Semver {
     ///         by ID was deposited for a given L2 token.
     mapping(address => mapping(address => mapping(uint256 => bool))) public deposits;
 
-    /// @custom:semver 1.1.2
-    /// @notice Constructs the L1ERC721Bridge contract.
+    /// @custom:semver 1.3.0
+    /// @notice Constructs the contract.
+    constructor() Semver(1, 3, 0) ERC721Bridge(Predeploys.L2_ERC721_BRIDGE) {
```
Comment (Inphi) [inline packages/contracts-bedrock/src/L1/L1ERC721Bridge.sol]: why skip 1.2.0; 1.1.2 -> 1.3.0?

**Review by Inphi — COMMENTED** [review]
looks good. one nit on contract semvers

File: packages/contracts-bedrock/src/L1/L1ERC721Bridge.sol (hunk 2 above)
Comment (tynes) [reply packages/contracts-bedrock/src/L1/L1ERC721Bridge.sol]: Good call

**Comment by tynes** [comment]
I've addressed all comments on this PR and believe its ready to be merged

**Review by refcell — APPROVED** [review]
Looks good now 🎉

**Review by clabby — APPROVED** [review]
LGTM

edit: 1 small nit

File: packages/contracts-bedrock/src/L2/L2ERC721Bridge.sol [hunk 3]
```diff
@@ -17,14 +18,18 @@ import { Semver } from "../universal/Semver.sol";
 ///         wait for the one-week challenge period to elapse before their Optimism-native NFT
 ///         can be refunded on L2.
 contract L2ERC721Bridge is ERC721Bridge, Semver {
-    /// @custom:semver 1.1.1
+    /// @custom:semver 1.3.0
```
Comment (clabby) [inline packages/contracts-bedrock/src/L2/L2ERC721Bridge.sol]: nit: This should be bumped to `1.2.0`

**Review by clabby — CHANGES_REQUESTED** [review]

**Review by clabby — APPROVED** [review]

**Comment by tynes** [comment]
@clabby addressed your comment

**Pull request merged**